core/insertion_marker_manager.ts

```diff
@@ -561,8 +555,31 @@ export class InsertionMarkerManager {
       // probably recreate the marker block (e.g. in getCandidate_), which is
       // called more often during the drag, but creating a block that often
       // might be too slow, so we only do it if necessary.
-      this.firstMarker = this.createMarkerBlock(this.topBlock);
-      insertionMarker = isLastInStack ? this.lastMarker : this.firstMarker;
+      if (isLastInStack && this.lastOnStack) {
+        if (this.lastMarker) {
+          eventUtils.disable();
+          try {
+            this.lastMarker.dispose();
+          } finally {
+            eventUtils.enable();
+          }
+        }
```

Comment:
Can you break this out into a sub procedure? E.g. `disposeInsertionMarker`

Reply:
Good call, done!